Dockerfile

```diff
@@ -1,5 +1,26 @@
 # Build the manager binary
-FROM registry.access.redhat.com/ubi8/go-toolset:1.20.10 as builder
+#FROM registry.access.redhat.com/ubi8/go-toolset:1.20.10 as builder
+
+# BEGIN -- workaround lack of go-toolset for golang 1.21
+
+# FROM registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:v1.21 AS golang
+FROM golang:1.21 AS golang
+FROM registry.access.redhat.com/ubi8/ubi:8.8 AS builder
```

Comment:
@dimakis Will this be okay in our build pipelines?

Reply:
This was based on a Dockerfile I got from @astefanutti for Kueue, but his file used a redhat internal  image AS `golang` that I didn't have access to.  I used the default `golang:1.21` to get unblocked, but its almost certainly not the right to use in the long run.

Reply:
no, we won't get away with that. we'll need to go back to a RH registry

Reply:
Is there a RH registry that has the a usable go 1.21 toolchain that is publicly accessible?  I couldn't find one.  If there is one, I'd be happy to change the docker file.

Reply:
That's fine for upstream, I think. This has to be overwritten anyway downstream.